### PR TITLE
Create bulldozer bot config

### PR DESCRIPTION
**Pull request by dansanduleac**
Configuration enables us to use the [bulldozer app](https://github.com/palantir/bulldozer).

### CODE REVIEW

File: .bulldozer.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: 1
+merge:
+  whitelist:
+    labels: ["merge when ready"]
+  blacklist:
+    labels: ["do not merge"]
+  method: squash
+  options:
+    squash:
+      body: pull_request_body
```
Comment (robert3005) [inline .bulldozer.yml]: How does this work for empty body?

Comment (dansanduleac) [reply .bulldozer.yml]: I don't know actually.. Isn't mentioned in the docs. https://github.com/palantir/bulldozer#caveats-and-notes

Comment (dansanduleac) [reply .bulldozer.yml]: From looking at another repo, the first line will always have the `title (PR number)`, and then the message (if any).

Example: https://github.com/palantir/conjure-java-runtime/commit/9cfbf297d5e3b7d6319e3f41bfc248e38fe26034

File: .bulldozer.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+version: 1
+merge:
```
Comment (robert3005) [inline .bulldozer.yml]: is approval enough given those rules?

File: .bulldozer.yml (hunk 1 above)
Comment (robert3005) [reply .bulldozer.yml]: ok

File: .bulldozer.yml (hunk 2 above)
Comment (dansanduleac) [reply .bulldozer.yml]: Yep - my understanding is that approval is decided as:
```
required checks you've configured in github for the repo pass
    && (some whitelist condition is met || there are no whitelist conditions)
    && no blacklist conditions are met
```

**Comment by dansanduleac** [comment]
The cool thing about this is, it will make it easy for people to do the right thing when they use `merge when ready` (squash, don't merge).
We should just make sure people don't use it when syncing upstream, as you don't want a squash commit then.

**Comment by robert3005** [comment]
I would prefer to go with just blacklist mode and require labeling upstream merges as do not merge

**Pull request merged**